web/hooks/github.go

```diff
@@ -81,15 +81,19 @@ func (wh GitHubWebHook) handlePush(payload *github.PushEvent) {
 		wh.logger.Debugf("Processing push event for %s", payload.GetRepo().GetName())
 		assignments := wh.extractAssignments(payload, course)
 		for _, assignment := range assignments {
-			runData := &ci.RunData{
-				Course:     course,
-				Assignment: assignment,
-				Repo:       repo,
-				CloneURL:   payload.GetRepo().GetCloneURL(),
-				CommitID:   payload.GetHeadCommit().GetID(),
-				JobOwner:   payload.GetSender().GetLogin(),
+			if repo.IsOfMatchingType(assignment.IsGroupLab) {
```

Comment:
I propose that we more clearly separate this logic. Let's instead make a separate:

```golang
case repo.IsGroupRepo():
    ...
```

With similar code as here. The reason for this is that then it will be easier to make future customizations for group repos etc. Perhaps we can just make a function for the common parts...

Reply:
Fixed, if I understood you correctly. Could you take a look?